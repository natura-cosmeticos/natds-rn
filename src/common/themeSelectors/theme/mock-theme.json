{
  "avatarSizes": {
    "tiny": {
      "size": 24,
      "fontSize": ".625rem"
    },
    "small": {
      "size": 32,
      "fontSize": ".875rem"
    },
    "standard": {
      "size": 40,
      "fontSize": "1rem"
    },
    "large": {
      "size": 48,
      "fontSize": "1rem"
    },
    "huge": {
      "size": 80,
      "fontSize": "1.5rem"
    }
  },
  "buttonSizes": {
    "small": {
      "paddingTop": 8,
      "paddingRight": 8,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "minHeight": 32
    },
    "medium": {
      "paddingTop": 12,
      "paddingRight": 16,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "minHeight": 40
    },
    "large": {
      "paddingTop": 16,
      "paddingRight": 16,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "minHeight": 48
    }
  },
  "fontSize": {
    "h1": {
      "fontSize": "6rem",
      "fontWeight": 300
    },
    "h2": {
      "fontSize": "3.75rem",
      "fontWeight": 300
    },
    "h3": {
      "fontSize": "3rem",
      "fontWeight": 400
    },
    "h4": {
      "fontSize": "2.125rem",
      "fontWeight": 400
    },
    "h5": {
      "fontSize": "1.5rem",
      "fontWeight": 400
    },
    "h6": {
      "fontSize": "1.25rem",
      "fontWeight": 600
    },
    "subtitle1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "subtitle2": {
      "fontSize": "0.875rem",
      "fontWeight": 500
    },
    "body1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "body2": {
      "fontSize": "0.875rem",
      "fontWeight": 400
    },
    "button": {
      "fontSize": "0.875rem",
      "fontWeight": 600
    },
    "caption": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    },
    "overline": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    }
  },
  "iconSizes": {
    "micro": 16,
    "tiny": 24,
    "small": 32,
    "standard": 48,
    "medium": 64,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "huge": 144,
    "hugex": 192,
    "hugexx": 256
  },
  "opacity": {
    "12": 0.12,
    "16": 0.16,
    "24": 0.24,
    "32": 0.32,
    "48": 0.48,
    "56": 0.56,
    "64": 0.64,
    "80": 0.8,
    "00": 0,
    "04": 0.04,
    "08": 0.08
  },
  "overlay": {
    "solidBlack": 0.56
  },
  "radius": {
    "none": 0,
    "circle": "50%",
    "small": 2,
    "medium": 4,
    "large": 8
  },
  "sizes": {
    "none": 0,
    "micro": 4,
    "tiny": 8,
    "small": 16,
    "standard": 24,
    "semi": 32,
    "semix": 40,
    "medium": 48,
    "mediumx": 56,
    "large": 64,
    "largex": 72,
    "largexx": 80,
    "largexxx": 88,
    "huge": 96,
    "hugex": 128,
    "hugexx": 144,
    "hugexxx": 192,
    "veryhuge": 256
  },
  "spacing": {
    "spacing": 8,
    "spacingNone": 0,
    "spacingMicro": 4,
    "spacingTiny": 8,
    "spacingSmall": 16,
    "spacingStandard": 24,
    "spacingSemi": 32,
    "spacingLarge": 48,
    "spacingHuge": 64
  },
  "colorTokens": {
    "colorPrimary": "#F4AB34",
    "colorOnPrimary": "#333333",
    "colorPrimaryLight": "#FEFDE8",
    "colorOnPrimaryLight": "#333333",
    "colorPrimaryDark": "#EF8426",
    "colorOnPrimaryDark": "#333333",
    "colorSecondary": "#FF6B0B",
    "colorOnSecondary": "#000000",
    "colorSecondaryLight": "#FFF8E1",
    "colorOnSecondaryLight": "#000000",
    "colorSecondaryDark": "#FF5808",
    "colorOnSecondaryDark": "#000000",
    "colorBackground": "#FAFAFA",
    "colorOnBackground": "#333333",
    "colorSurface": "#FFFFFF",
    "colorOnSurface": "#333333",
    "colorHighlight": "#000000",
    "colorHighEmphasis": "#333333",
    "colorMediumEmphasis": "#777777",
    "colorLowEmphasis": "#BBBBBB",
    "colorLink": "#227BBD",
    "colorOnLink": "#FFFFFF",
    "colorSuccess": "#569A32",
    "colorOnSuccess": "#FFFFFF",
    "colorWarning": "#FCC433",
    "colorOnWarning": "#333333",
    "colorAlert": "#E74627",
    "colorOnAlert": "#FFFFFF"
  },
  "elevation": {
    "1": {
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 1, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 1
    },
    "2": {
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 2, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 2
    },
    "3": {
      "elevation": 3,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 3, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 4
    },
    "4": {
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 4, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 5
    },
    "5": {
      "elevation": 5,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 6, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "6": {
      "elevation": 6,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 8, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "7": {
      "elevation": 7,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 9, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 12
    },
    "8": {
      "elevation": 8,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 12, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 17
    },
    "9": {
      "elevation": 9,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 16, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 24
    },
    "10": {
      "elevation": 10,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 24, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 38
    }
  }
}
