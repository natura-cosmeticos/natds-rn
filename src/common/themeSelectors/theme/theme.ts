import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
