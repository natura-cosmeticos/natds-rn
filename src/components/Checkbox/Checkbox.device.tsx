import { storiesOf } from '@storybook/react-native';
import {
  Default,
  Indeterminate,
  Disabled,
  Interactive,
} from './Checkbox.stories';

storiesOf('Checkbox', module)
  .add('Default', Default)
  .add('Indeterminate', Indeterminate)
  .add('Disabled', Disabled)
  .add('Interactive', Interactive);
