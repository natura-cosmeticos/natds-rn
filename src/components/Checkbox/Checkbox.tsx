import React from 'react';
import { TouchableRipple } from '../TouchableRipple';
import {
  Container, Box, Check, Label,
} from './Checkbox.styles';

export type CheckboxColors = 'primary' | 'secondary';

export interface CheckboxProps {
  /**
   * Checkbox color: `primary` | `secondary`
   */
  color?: CheckboxColors;
  /**
   * Called when the touch is released, but not if cancelled
   */
  onPress?: (value: string) => void;
  /**
   * Programmatically tell if the component is selected
   */
  selected?: boolean;
  /**
   * Optional property that tells if the select is disabled,
   * if so, the user cannot click on this item.
   */
  disabled?: boolean;
  /**
   * Optional label that appears to the right of the component,
   * if the user clicks that label, the onPress function will be called
   */
  label?: string;
  /**
   * The value of the component.
   */
  value?: string;
  /**
   * Optional testID
   */
  testID?: string;
}

export const Checkbox = ({
  color = 'primary',
  onPress,
  selected,
  disabled = false,
  label,
  value = '',
  testID = 'checkbox',
}: CheckboxProps) => {
  const onPressWithValue = () => {
    if (onPress) {
      onPress(value);
    }
  };
  const checkbox = (
    <TouchableRipple
      size={20}
      color={color}
      onPress={onPressWithValue}
      disabled={disabled}
      testID={testID}
    >
      <Box disabled={disabled} color={color} selected={selected} testID={`${testID}-circle`}>
        {
          selected
            ? <Check />
            : null
        }
      </Box>
    </TouchableRipple>
  );

  if (label) {
    return (
      <Container disabled={disabled} testID={`${testID}-label`} onPress={onPressWithValue}>
        {checkbox}
        <Label disabled={disabled}>{label}</Label>
      </Container>
    );
  }

  return checkbox;
};
