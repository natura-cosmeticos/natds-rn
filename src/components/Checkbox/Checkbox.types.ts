/* eslint-disable max-len */
import { AccessibilityProps } from 'react-native';

/**
 * @deprecated this prop is no longer necessary as this component should always be used with the primary color.
 */
export type CheckboxColors = 'primary' | 'secondary';

type AccessibilityCheckboxProps = Pick<AccessibilityProps,
  'accessibilityLabel' |
  'accessibilityHint' |
  'accessibilityRole' |
  'accessibilityState'
>

export interface CheckboxProps extends AccessibilityCheckboxProps {
  /**
   * <span style="color:red">*Deprecated, this prop is no longer necessary as this component should always be used with the primary color.*</span>
   * Define what the color will be, according to the corresponding values.
   *
   * - Available colors: `primary`, `secondary`.
   *
   * @default `primary`
   */
  color?: CheckboxColors;
  /**
   * Called when the touch is released, but not if cancelled.
   */
  onPress?: (value: string) => void;
  /**
   * Programmatically tell if the component is selected.
   * @default false
   */
  selected?: boolean;
  /**
   * Optional property that tells if the select is disabled,
   * if so, the user cannot click on this item.
   * @default false
   */
  disabled?: boolean;
  /**
   * Indeterminate checkboxes are displayed with a dash inside
   * the box, being visually different than default checkboxes.
   * @default false
   */
  indeterminate?: boolean;
  /**
   * Optional label that appears to the right of the component,
   * if the user clicks that label, the onPress function will be called.
   */
  label?: string;
  /**
   * This value determines the identity of the component and is returned in the onPress function.
   */
  value?: string;
  /**
   * Optional testID.
   */
  testID?: string;
}
