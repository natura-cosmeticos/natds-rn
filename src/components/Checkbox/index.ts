export { Checkbox } from './Checkbox';
export { CheckboxProps } from './Checkbox.types';
