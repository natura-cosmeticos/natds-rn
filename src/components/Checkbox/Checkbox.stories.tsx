/* eslint-disable max-lines */
/* eslint-disable max-len */
import React, { useState } from 'react'
import { boolean, select, text } from '@storybook/addon-knobs'

import { Checkbox } from './Checkbox'
import { StoryContainer, StoryWrapper } from '../../common/HelperComponents/StoryContainer'
import { CheckboxColors } from './Checkbox.types'

const description = () => `
- - -

  ### NOTE:
  This component is available in the following variants:

  - ✅ **Standard**
  - ✅ **Indeterminate**

With the following attribute status:

  - ✅ **Disabled**

- - -
`

export default {
  component: Checkbox,
  parameters: {
    componentSubtitle: 'Selection controls allow the user to select options.',
    docs: {
      extractComponentDescription: description
    }
  },
  title: 'Components|Checkbox'
}

const colorTypes = {
  primary: 'primary',
  secondary: 'secondary'
}

export const Default = () => {
  const [isSelected, setIsSelected] = useState(true)

  return (
    <StoryContainer title="Standard">
      <Checkbox
        label="Natura Design System"
        onPress={() => setIsSelected(!isSelected)}
        selected={isSelected}
      />
    </StoryContainer>
  )
}

export const Variants = () => {
  const [standard, setStandard] = useState(true)
  const [indeterminate, setIndeterminate] = useState(true)

  return (
    <StoryWrapper title="Variants">
      <StoryContainer title="Standard">
        <Checkbox
          label="Natura Design System"
          onPress={() => setStandard(!standard)}
          selected={standard}
        />
      </StoryContainer>
      <StoryContainer title="Indeterminate">
        <Checkbox
          indeterminate
          label="Natura Design System"
          onPress={() => setIndeterminate(!indeterminate)}
          selected={indeterminate}
        />
      </StoryContainer>
      <StoryContainer title="Disabled selected">
        <Checkbox
          disabled
          label="Natura Design System"
          selected
          value="3"
        />
      </StoryContainer>
      <StoryContainer title="Disabled indeterminate">
        <Checkbox
          disabled
          indeterminate
          label="Natura Design System"
          selected
          value="2"
        />
      </StoryContainer>
      <StoryContainer title="Disabled unselected">
        <Checkbox
          disabled
          label="Natura Design System"
          selected={false}
          value="1"
        />
      </StoryContainer>
    </StoryWrapper>
  )
}

export const Interactive = () => (
  <StoryContainer title="Interactive">
    <Checkbox
      color={select('Color', colorTypes, 'primary') as CheckboxColors}
      indeterminate={boolean('Indeterminate', false)}
      selected={boolean('Selected', true)}
      disabled={boolean('Disabled', false)}
      onPress={() => null}
      label={text('Label', 'Natura Design System')}
      value={text('Value', 'my-label')}
    />
  </StoryContainer>
)
