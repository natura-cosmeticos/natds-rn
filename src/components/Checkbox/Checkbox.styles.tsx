
import styled from 'styled-components/native';
import {
  Theme,
  getColorLowEmphasis,
  getColorMediumEmphasis,
  getColorOnBackground,
  getColorPrimary,
  getColorSecondary,
  getSpacingTiny,
} from '../../common/themeSelectors';
import { CheckboxColors } from './Checkbox';

function getBoxColor(
  selected = false,
  disabled: boolean,
  color: CheckboxColors,
  theme: Theme,
) {
  if (disabled) {
    return getColorLowEmphasis(theme);
  }

  if (selected) {
    return color === 'secondary' ? getColorSecondary(theme) : getColorPrimary(theme);
  }

  return getColorMediumEmphasis(theme);
}

export const Container = styled.TouchableOpacity.attrs({
  activeOpacity: 1,
})`
  flex-direction: row;
  align-items: center;
`;

export const Label = styled.Text<{ theme: Theme; disabled: boolean }>(({ theme, disabled }) => ({
  color: disabled ? getColorLowEmphasis(theme) : getColorOnBackground(theme),
  fontSize: 14,
  letterSpacing: 0.22,
  marginLeft: getSpacingTiny(theme),
}));


export const Box = styled.View<{
  theme: Theme;
  color: CheckboxColors;
  selected?: boolean;
  disabled: boolean;
}>(({
  theme,
  color,
  selected,
  disabled,
}) => ({
  alignItems: 'center',
  backgroundColor: selected ? getBoxColor(selected, disabled, color, theme) : 'transparent',
  borderColor: getBoxColor(selected, disabled, color, theme),
  borderRadius: 3,
  borderWidth: 2,
  height: 20,
  justifyContent: 'center',
  width: 20,
}));

export const Wrapper = styled.View(({
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  padding: 8,
}));
