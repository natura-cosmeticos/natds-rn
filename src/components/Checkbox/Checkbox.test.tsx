import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { fireEvent, render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import theme from '../../common/themeSelectors/theme/mock-theme.json';
import { Checkbox, CheckboxProps } from './Checkbox';
// import * as TouchableRipple from '../TouchableRipple/TouchableRipple';

jest.mock('../TouchableRipple/TouchableRipple');
jest.mock(
  'react-native/Libraries/Components/Touchable/TouchableOpacity',
  () => 'TouchableOpacity',
);

const renderCheckbox = (fn, props?: Omit<CheckboxProps, 'theme'>) => (fn(
  <ThemeProvider theme={theme}>
    <Checkbox {...props} />
  </ThemeProvider>,
));

describe('Checkbox component', () => {
  it('should render checkbox correctly', () => {
    const checkbox = renderCheckbox(renderer.create).toJSON();

    expect(checkbox).toMatchSnapshot();
  });

  it('should render checkbox selected correctly', () => {
    const checkbox = renderCheckbox(renderer.create, { selected: true }).toJSON();

    expect(checkbox).toMatchSnapshot();
  });

  it('should render checkbox with label correctly', () => {
    const checkbox = renderCheckbox(renderer.create, { label: 'My Label' }).toJSON();

    expect(checkbox).toMatchSnapshot();
  });

  it('should render checkbox selected but disabled correctly', () => {
    const checkbox = renderCheckbox(renderer.create, {
      disabled: true,
      selected: true,
    }).toJSON();

    expect(checkbox).toMatchSnapshot();
  });

  it('should render checkbox selected but disabled with label correctly', () => {
    const checkbox = renderCheckbox(renderer.create, {
      disabled: true,
      label: 'My Label',
      selected: true,
    }).toJSON();

    expect(checkbox).toMatchSnapshot();
  });


  it('should call the onPress function when the user touches the checkbox', () => {
    const onPress = jest.fn();
    const { queryByTestId } = renderCheckbox(render, {
      onPress,
      value: 'checkbox',
    });

    const checkbox = queryByTestId('checkbox');

    fireEvent.press(checkbox);

    expect(onPress).toHaveBeenCalledWith('checkbox');
  });

  it('should call the onPress function when the user touches the checkbox label', () => {
    const onPress = jest.fn();
    const { queryByTestId } = renderCheckbox(render, {
      label: 'My label',
      onPress,
      value: 'checkbox-label',
    });

    const label = queryByTestId('checkbox-label');

    fireEvent.press(label);

    expect(onPress).toHaveBeenCalledWith('checkbox-label');
  });

  it('should render check inside box when selected is true', () => {
    const onPress = jest.fn();
    const { queryByTestId } = renderCheckbox(render, {
      onPress,
      selected: true,
      value: 'checkbox',
    });

    const checkbox = queryByTestId('checkbox');

    expect(checkbox.props.children).not.toBeNull();
  });

  it('should render text inside label container when label is different from null/undefined', () => {
    const onPress = jest.fn();
    const { queryByTestId } = renderCheckbox(render, {
      label: 'My Label',
      onPress,
      value: 'checkbox',
    });

    const checkboxLabel = queryByTestId('checkbox-label');

    expect(checkboxLabel).not.toBeNull();
  });
});
