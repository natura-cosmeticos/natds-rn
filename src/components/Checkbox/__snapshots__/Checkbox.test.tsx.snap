// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component render should render checkbox correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="checkbox"
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 50,
      }
    }
    testID="touchable-ripple"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        color="primary"
        disabled={false}
        selected={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#777777",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="checkbox-box"
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Checkbox component render should render checkbox disabled correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="checkbox"
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 50,
      }
    }
    testID="touchable-ripple"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        color="primary"
        disabled={true}
        selected={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#bbbbbb",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="checkbox-box"
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Checkbox component render should render checkbox indeterminate selected but disabled correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="checkbox"
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 50,
      }
    }
    testID="touchable-ripple"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        color="primary"
        disabled={true}
        selected={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#bbbbbb",
              "borderColor": "#bbbbbb",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="checkbox-box"
      >
        <Text
          accessibilityRole="image"
          color="surface"
          size="small"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "natds-icons",
                "fontSize": 16,
              },
            ]
          }
          testID="natds-icon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Checkbox component render should render checkbox indeterminate selected correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="checkbox"
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 50,
      }
    }
    testID="touchable-ripple"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        color="primary"
        disabled={false}
        selected={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f4ab34",
              "borderColor": "#f4ab34",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="checkbox-box"
      >
        <Text
          accessibilityRole="image"
          color="surface"
          size="small"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "natds-icons",
                "fontSize": 16,
              },
            ]
          }
          testID="natds-icon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Checkbox component render should render checkbox selected but disabled correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="checkbox"
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 50,
      }
    }
    testID="touchable-ripple"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        color="primary"
        disabled={true}
        selected={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#bbbbbb",
              "borderColor": "#bbbbbb",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="checkbox-box"
      >
        <Text
          accessibilityRole="image"
          color="surface"
          size="small"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "natds-icons",
                "fontSize": 16,
              },
            ]
          }
          testID="natds-icon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Checkbox component render should render checkbox selected correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="checkbox"
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 50,
      }
    }
    testID="touchable-ripple"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        color="primary"
        disabled={false}
        selected={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f4ab34",
              "borderColor": "#f4ab34",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="checkbox-box"
      >
        <Text
          accessibilityRole="image"
          color="surface"
          size="small"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "natds-icons",
                "fontSize": 16,
              },
            ]
          }
          testID="natds-icon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;
